File: src/main/java/com/indiepost/service/ContributorService.java
package com.indiepost.service;

import com.indiepost.dto.ContributorDto;
import com.indiepost.enums.Types;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;


public interface ContributorService {

    ContributorDto findOne(Long id);

    ContributorDto save(ContributorDto contributor);

    Long delete(ContributorDto contributor);

    Long deleteById(Long id);

    int count(Types.ContributorType type);

    int count();

    Page<ContributorDto> find(Types.ContributorType type, Pageable pageable);

    Page<ContributorDto> find(Pageable pageable);
}

File: src/main/kotlin/com/indiepost/service/ContributorService.kt
package com.indiepost.service

import com.indiepost.dto.ContributorDto
import com.indiepost.enums.Types
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable


interface ContributorService {

    fun findOne(id: Long): ContributorDto?

    fun save(dto: ContributorDto): ContributorDto

    fun delete(dto: ContributorDto): Long?

    fun deleteById(id: Long): Long?

    fun count(type: Types.ContributorType): Int

    fun count(): Int

    fun find(type: Types.ContributorType, pageable: Pageable): Page<ContributorDto>

    fun find(pageable: Pageable): Page<ContributorDto>
}

